src/Entity/ElectedRepresentative/PoliticalFunction.php

```diff
@@ -0,0 +1,194 @@
+     *     message="La date de fin peut être saisie que dans le cas où la fonction est en cours."
+     * )
+     */
+    private $finishAt;
+
+    /**
+     * @var ElectedRepresentative
+     *
+     * @ORM\ManyToOne(targetEntity="ElectedRepresentative", inversedBy="politicalFunctions")
+     * @ORM\JoinColumn(nullable=false, onDelete="CASCADE")
+     *
+     * @Assert\NotBlank
+     */
+    private $electedRepresentative;
+
+    public function __construct(
+        string $name = null,
+        string $clarification = null,
+        string $geographicalArea = null,
+        ElectedRepresentative $electedRepresentative = null,
+        bool $onGoing = true,
+        \DateTime $beginAt = null,
+        \DateTime $finishAt = null
+    ) {
+        $this->name = $name;
+        $this->clarification = $clarification;
+        $this->geographicalArea = $geographicalArea;
+        $this->electedRepresentative = $electedRepresentative;
+        $this->onGoing = $onGoing;
+        $this->beginAt = $beginAt;
+        $this->finishAt = $finishAt;
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): void
+    {
+        if (!PoliticalFunctionNameEnum::isValid($name)) {
+            throw new BadPoliticalFunctionNameException(sprintf('The political function name "%s" is invalid', $name));
+        }
+
+        $this->name = $name;
+    }
+
+    public function getClarification(): ?string
+    {
+        return $this->clarification;
+    }
+
+    public function setClarification(?string $clarification = null): void
+    {
+        $this->clarification = $clarification;
+    }
+
+    public function getGeographicalArea(): ?string
+    {
+        return $this->geographicalArea;
+    }
+
+    public function setGeographicalArea(string $geographicalArea): void
+    {
+        $this->geographicalArea = $geographicalArea;
+    }
+
+    public function isOnGoing(): bool
+    {
+        return $this->onGoing;
+    }
+
+    public function setOnGoing(bool $onGoing): void
+    {
+        $this->onGoing = $onGoing;
+    }
+
+    public function getBeginAt(): ?\DateTime
+    {
+        return $this->beginAt;
+    }
+
+    public function setBeginAt(\DateTime $beginAt): void
+    {
+        $this->beginAt = $beginAt;
+    }
+
+    public function getFinishAt(): ?\DateTime
+    {
+        return $this->finishAt;
+    }
+
+    public function setFinishAt(?\DateTime $finishAt = null): void
+    {
+        $this->finishAt = $finishAt;
+    }
+
+    public function setElectedRepresentative(ElectedRepresentative $electedRepresentative): void
+    {
+        $this->electedRepresentative = $electedRepresentative;
+    }
+
+    public function __toString(): string
+    {
+        return array_search($this->name, PoliticalFunctionNameEnum::CHOICES);
```

Comment:
`(string) array_search(...)` as the function can return `false|int|string` types